### PR TITLE
fix syndication settings to not write on read

### PR DESCRIPTION
**Pull request by gforcada**
Backport CSRF fix to 4.3.x

### CODE REVIEW

**Comment by gforcada** [comment]
Jenkins: http://jenkins.plone.org/job/pull-request-5.0/455/

File: Products/CMFPlone/browser/syndication/settings.py [hunk 1]
```diff
@@ -17,19 +17,30 @@ class FeedSettings(object):
...
+        if self.needs_saving:
+            self.annotations[FEED_SETTINGS_KEY] = self._metadata
+
     def __setattr__(self, name, value):
-        if name in ('context', '_metadata', 'site_settings'):
+        if name in ('context', '_metadata', 'site_settings', 'annotations',
+                    'needs_saving'):
             self.__dict__[name] = value
         else:
             self._metadata[name] = value
```
Comment (vangheem) [inline Products/CMFPlone/browser/syndication/settings.py]: You missed just one part of the back-port. need to call self._set()

https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/browser/syndication/settings.py#L47

Comment (gforcada) [inline Products/CMFPlone/browser/syndication/settings.py]: Fixed

**Comment by gforcada** [comment]
Running again: http://jenkins.plone.org/job/pull-request-4.3/126

**Comment by gforcada** [comment]
And finally tests passes http://jenkins.plone.org/job/pull-request-4.3/129

@vangheem should we merge it?

**Pull request merged**